src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php

```diff
@@ -81,6 +82,9 @@
         ->set('argument_resolver.variadic', VariadicValueResolver::class)
             ->tag('controller.argument_value_resolver', ['priority' => -150])
 
+        ->set('argument_resolver.query_parameter_value_resolver', QueryParameterValueResolver::class)
+            ->tag('controller.pinned_value_resolver', ['name' => QueryParameterValueResolver::class])
```

Comment:
why `controller.pinned_value_resolver` and not `controller.argument_value_resolver` :thinking:  ?

Reply:
See #48992

Reply:
mentioned PR already changed little bit, so we can use old tag name and just add `name`

Reply:
We want to use the new tag name because we want the resolver to be called only when the attribute is explicitly set on a parameter - and not be part of the regular chain of resolvers.